pkg/apis/databases/v1alpha4/connection.go

```diff
@@ -277,27 +268,38 @@ func (d *Database) getConnectionFromURI(ctx context.Context) (string, string, er
 	if err != nil {
 		return "", "", errors.Wrap(err, "failed to get database type")
 	}
-	var valueOrValueFrom ValueOrValueFrom
+	var valueOrValueFrom *ValueOrValueFrom
 	if driver == "postgres" {
-		valueOrValueFrom = d.Spec.Connection.Postgres.URI
+		valueOrValueFrom = &d.Spec.Connection.Postgres.URI
 	} else if driver == "cockroachdb" {
-		valueOrValueFrom = d.Spec.Connection.CockroachDB.URI
+		valueOrValueFrom = &d.Spec.Connection.CockroachDB.URI
 	} else if driver == "cassandra" {
 		return "", "", errors.New("reading cassandra connecting from uri is not supported")
 	} else if driver == "mysql" {
-		valueOrValueFrom = d.Spec.Connection.Mysql.URI
+		valueOrValueFrom = &d.Spec.Connection.Mysql.URI
 	} else if driver == "rqlite" {
-		valueOrValueFrom = d.Spec.Connection.RQLite.URI
+		valueOrValueFrom = &d.Spec.Connection.RQLite.URI
 	} else if driver == "timescaledb" {
-		valueOrValueFrom = d.Spec.Connection.TimescaleDB.URI
+		valueOrValueFrom = &d.Spec.Connection.TimescaleDB.URI
 	}
 
 	value, err := d.getValueFromValueOrValueFrom(ctx, driver, valueOrValueFrom)
 	return driver, value, err
 }
 
+func (d *Database) addQueryStringParameter(ctx context.Context, driver string, queryParams *[]string, label string, key string, valueOrValueFrom *ValueOrValueFrom) error {
+	if !valueOrValueFrom.IsEmpty() {
+		value, err := d.getValueFromValueOrValueFrom(ctx, driver, valueOrValueFrom)
+		if err != nil {
+			return errors.Wrap(err, fmt.Sprintf("failed to read %s", label))
+		}
+		*queryParams = append(*queryParams, fmt.Sprintf("%s=%s", key, value))
+	}
+	return nil
+}
+
 // getValueFromValueOrValueFrom returns the resolved value, or an error
-func (d Database) getValueFromValueOrValueFrom(ctx context.Context, driver string, valueOrValueFrom ValueOrValueFrom) (string, error) {
+func (d *Database) getValueFromValueOrValueFrom(ctx context.Context, driver string, valueOrValueFrom *ValueOrValueFrom) (string, error) {
```

Comment:
If `valueOrValueFrom` is going to be a pointer, there should be a nil check. But it really doesn't need to be a pointer with the current implementation.

Reply:
Agree that it doesn't need to be, but this avoids copying the value unnecessarily. I'll add a nil check.